packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
Yes, as I've mentioned several times, there's a class of permanent errors which are good reasons to stop trying to open a connection:

- The server tells us the subscription does not exist
- The server tells us the topic does not exist
- The server tells us we don't have permission to use the subscription
- We fail to open a connection for a long time (five minutes)

Versus this class of errors which are not permanent which do need to be retried:

- Deadline exceeded
- Quota exhausted
- Any miscellaneous "server error", like an HTTP 500